File: Difficulty: Easy/Reverse Words/reverse-words.py
class Solution:
    def reverseWords(self, s):
        # code here
        parts = s.split('.')
        
        words = [w for w in parts if w]
        
        i = 0
        j = len(words)-1
        
        while i < j:
            words[i], words[j] = words[j], words[i]
            i += 1
            j -= 1
            
        return ".".join(words)
        

File: Difficulty: Easy/Reverse Words/reverse-words.java
//{ Driver Code Starts
import java.io.*;
import java.lang.*;
import java.util.*;

class GFG {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            String s = sc.next();
            Solution obj = new Solution();
            System.out.println(obj.reverseWords(s));
            t--;
        }
    }
}

// } Driver Code Ends



class Solution {
    // Function to reverse words in a given string.
    String reverseWords(String str) {
        // code here
        String arr[] = str.split("\\.");  
        StringBuilder ans = new StringBuilder();

        for (int i = arr.length - 1; i >= 0; i--) {
            ans.append(arr[i]);
            if (i != 0) {
                ans.append(".");
            }
        }

        return ans.toString();
    }
}